From cffa5e75506ae65583f5b7b0909705559740d74f Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Tue, 05 Jul 2022 17:11:10 +0300
Subject: [PATCH] [SW-94323] gaudi2: watch dog de-assertion in erom recovery.

At PRSTn CPLD asserts GPIO62 and starts WD timer.
Since de-assertion of this gpio is not done in erom recovery
even after successful firmware, ROM doesn't load and verify new
firmware from SPI as it enters to recovery due to GPIO2 assertion.
Hence de-asserting this gpio in erom recovery and after successful
firmware update the ROM should load, verify and execute new firmware
from spi flash.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I303d0d795c9eb3d18c79ebbbe6ea3f6084aa3d85
---

diff --git a/app/ext_rom/erom_recovery.c b/app/ext_rom/erom_recovery.c
index 5d1f4e4..16aa192 100644
--- a/app/ext_rom/erom_recovery.c
+++ b/app/ext_rom/erom_recovery.c
@@ -132,6 +132,8 @@
 	preboot_security_sp_sram_init();
 	preboot_dev_enable_caps();
 
+	preboot_clear_wd_gpio();
+
 	cpu_boot_sts_set(CPU_BOOT_STATUS_WAITING_FOR_BOOT_FIT);
 
 	preboot_puts("In recovery mode\n");
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index dc3fca3..0a6bfd7 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1041,7 +1041,7 @@
 	mask = BIT(WD_ASSERTION_GPIO);
 	regmap_writel(mask, mmPSOC_GPIO3_OUTENSET);
 	val  = regmap_readl(mmPSOC_GPIO3_DATAOUT);
-	val &= ~mask;
+	val |= mask;
 	regmap_writel(val, mmPSOC_GPIO3_DATAOUT);
 }
 
